infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -146,10 +140,37 @@
         this.initializationTaskThread.start();
     }
 
+    private void tryImmediateRecovery() {
+        log.info("Attempting immediate recovery");
+        for(int i = 0; i < 3; i++) {
```

Comment:
Since the layout response and recovery can be sometimes late due to miscellaneous issues like infra, Can we retry more times (say 5 or 10)?

Reply:
Usually 3 attempts for somethingn like attemptConsensus workflow is fine. We have some wait between them too, so I think it should be ok